src/components/Modal/src/Modal.vue

```diff
@@ -7,6 +7,55 @@
 	</div>
 </template>
 
+<script>
+import modalApi from './modal-api';
+
+export default {
+	inject: {
+		modalApi,
+	},
+
+	props: {
+		/**
+		 * Close the modal on ESC
+		 */
+		closeOnEsc: {
+			type: Boolean,
+			default: false,
+		},
+	},
+
+	mounted() {
+		this.addEvents();
+	},
+
+	destroyed() {
+		this.removeEvents();
+	},
+
+	methods: {
+		addEvents() {
+			if (this.closeOnEsc) {
+				document.addEventListener('keyup', this.handleEscKey);
+			}
+		},
+
+		removeEvents() {
+			if (this.closeOnEsc) {
+				document.removeEventListener('keyup', this.handleEscKey);
+			}
+		},
```

Comment:
You can replace the ~20 lines above using [vue-pseudo-window](https://github.com/privatenumber/vue-pseudo-window) which is a dep we already include in Maker. After removing these lines you'd just add this to the template above:

```vue
<pseudo-window
	document
	@keyup="handleEscKey"
/>
```

Reply:
`keyup.esc="handleEscKey"`